src/detectAdBlocker.ts

```diff
@@ -0,0 +1,65 @@
+/**
+Detect whether or not the user has an ad blocking extension enabled.
+A few ad blockers are not detectable with this approach e.g. Safari / Adblock
+Code inspired by just-detect-adblock's: https://git.io/JgL4L
+*/
+/*istanbul ignore file -- adElementBlocked can't be tested without patching each of the properties of
+HTMLElement.prototype that it accesses, defeating the purpose of the test! */
+
+let adBlockInUse: boolean | undefined = undefined;
+
+function adElementBlocked(ad: HTMLElement): boolean {
+	if (
+		ad.offsetParent === null ||
+		ad.offsetHeight === 0 ||
+		ad.offsetLeft === 0 ||
+		ad.offsetTop === 0 ||
+		ad.offsetWidth === 0 ||
+		ad.clientHeight === 0 ||
+		ad.clientWidth === 0
+	)
+		return true;
+	const adStyles = window.getComputedStyle(ad);
+
+	if (adStyles.getPropertyValue('display') === 'none') return true;
+	if (adStyles.getPropertyValue('visibility') === 'hidden') return true;
+
+	const mozBindingProp = adStyles.getPropertyValue('-moz-binding');
+	if (mozBindingProp.includes('about:')) return true;
+
+	return false;
+}
+
+export function isAdBlockInUse(): Promise<boolean> {
```

Comment:
Could we add a comment indicating how reliable this is?

For example, `false` is maybe and `true` is definitely.